zebra-script/src/lib.rs

```diff
@@ -167,39 +173,33 @@ mod tests {
             .branch_id()
             .expect("Blossom has a ConsensusBranchId");
 
-        is_valid(transaction, branch_id, (input_index, output))?;
+        let verifier = super::PrecomputedTransaction::new(&transaction);
+        verifier.is_valid(branch_id, (input_index, output))?;
 
         Ok(())
     }
 
     #[test]
-    fn verify_valid_script() -> Result<()> {
+    fn fail_invalid_script() -> Result<()> {
         zebra_test::init();
 
-        let coin = i64::pow(10, 8);
-        let script_pub_key = &*SCRIPT_PUBKEY;
-        let amount = 212 * coin;
-        let tx_to = &*SCRIPT_TX;
-        let n_in = 0;
-        let branch_id = 0x2bb40e60;
-
-        verify_script(script_pub_key, amount, tx_to, n_in, branch_id)?;
-
-        Ok(())
-    }
-
-    #[test]
-    fn dont_verify_invalid_script() -> Result<()> {
-        zebra_test::init();
-
-        let coin = i64::pow(10, 8);
-        let script_pub_key = &*SCRIPT_PUBKEY;
-        let amount = 212 * coin;
-        let tx_to = &*SCRIPT_TX;
-        let n_in = 0;
-        let branch_id = 0x2bb40e61;
+        let transaction =
+            SCRIPT_TX.zcash_deserialize_into::<Arc<zebra_chain::transaction::Transaction>>()?;
+        let coin = u64::pow(10, 8);
+        let amount = 211 * coin;
+        let output = transparent::Output {
+            value: amount.try_into()?,
+            lock_script: transparent::Script(SCRIPT_PUBKEY.clone()),
+        };
+        let input_index = 0;
+        let branch_id = Blossom
+            .branch_id()
+            .expect("Blossom has a ConsensusBranchId");
 
-        verify_script(script_pub_key, amount, tx_to, n_in, branch_id).unwrap_err();
+        let verifier = super::PrecomputedTransaction::new(&transaction);
```

Comment:
Can we also have a separate test that re-uses a pre-computed script?

Let's test the following input combinations:
* fail / fail
* fail / success
* success / fail